samples/sensor/die_temp_polling/boards/sam4s_xplained.overlay

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2023, Basalte bv
+ */
+
+#include <zephyr/dt-bindings/adc/adc.h>
+
+/ {
+	aliases {
+		die-temp0 = &die_temp;
+	};
+
+        die_temp: die-temp {
+		status = "okay";
+		compatible = "zephyr,adc-thermistor";
+		io-channels = <&adc0 15>;
+		temperature-lookup-table = <
+			105000   1806
+			27000    1440
+			(-40000) 1125
+		>;
```

Comment:
There is some inconsistency about lookup table size:

```c
#define ADC_THERMISTOR_DEFINE(inst)								\
	BUILD_ASSERT(DT_INST_PROP_LEN_OR(inst, temperature_lookup_table, 0) % 2 == 0,		\
		     "Temperature lookup table needs an even size");	
```

Please, clarify!

Reply:
This lookup table has 6 elements, so it is consistent. Note that the elements are formatted in pairs to make it clearer which temperature is tied to each voltage.